lib/streamlit/runtime/app_session.py

```diff
@@ -443,8 +443,8 @@ def _handle_scriptrunner_event_on_main_thread(
         """
 
         assert (
-            threading.main_thread() == threading.current_thread()
-        ), "This function must only be called on the main thread"
+            self._event_loop == asyncio.get_running_loop()
+        ), "This function must only be called on the eventloop thread"
```

Comment:
This is the only functional change in the PR. Everything else is test updates and a function rename.